File: CDAC/DS_Team_6/Queue.cpp
#include<iostream>
#include<stdio.h>
using namespace std;
class Queue{
    private:
        int* arr;
        int size;
        int front;
        int rear;

    public: 
        Queue(int size){
            this->size = size;
            arr = new int[size];
            front = -1;
            rear = -1;
        }

        bool isEmpty(){
            if(front == -1 || front > rear){
                return true;
            }else{
                return false;
            }
        }

        bool isFull(){
            if(rear == size-1){
                return true;
            }else{
                return false;
            }
        }

        void enqueue(int data){
            if(!isFull()){
                if(front == -1)
                    front++;
                rear++;
                arr[rear] = data;
                cout<<data<<" enqueued sucessfully!!!"<<endl;
            }else{
                cout<<"Queue is full!!!"<<endl;
            }
        }

        void dequeue(){
            if(!isEmpty()){
                cout<<"Dequeued element is "<<arr[front++]<<endl;
            }else{
                cout<<"Queue is empty!!!"<<endl;
            }
        }

        void display(){
            if(!isEmpty()){
                cout<<"Elements in the queue : ";
                for(int i=front;i<=rear;i++){
                    cout<<arr[i]<<" ";
                }
                cout<<endl;
            }else{
                cout<<"Queue is empty!!!"<<endl;
            }
        }
};
int main(){
    int size,data,choice;
    cout<<"Enter the size of data: ";
    cin>>size;
    Queue q(size);
    while(1){
        cout<<"1. Enqueue\n2. Dequeue\n3. Display\n4. Exit"<<endl;
        cout<<"Enter your choice: ";
        cin>>choice;
        switch(choice){
            case 1:
                cout<<"Enter the data: ";
                cin>>data;
                q.enqueue(data);
                break;
            case 2:
                q.dequeue();
                break;
            case 3:
                q.display();
                break;
            case 4:
                exit(0);
            default:
                cout<<"Enter the correct choice!!!"<<endl;
        }
    }
}

File: CDAC/edac_data_structure/src/edac_data_structure/Queue.java
package edac_data_structure;

import java.util.Scanner;

public class Queue {

	private int arr[];
	private int size;
	private int front;
	private int rear;
	Queue(){
		arr = new int[0];
		size=0;
		front = rear = -1;
	}
	
	Queue(int size) {
		this.size=size;
		arr = new int[size];
		front = rear = -1;
	}
	
	void enqueue(int data) {
		if(rear == size-1) {
			System.out.println("Queue is full!!!");
		}
		else if(front == -1 && rear == -1) {
			front++;
			rear++;
			arr[rear]=data;
			System.out.println(data+" added successfully!!!");
		}else {
			rear++;
			arr[rear]=data;
			System.out.println(data+" added successfully!!!");
		}
	}
	
	void dequeue() {
		if(front == -1 && rear == -1)
			System.out.println("Queue is empty!!!");
		else if(front>rear) {
			System.out.println("Queue is empty!!!");
		}else {
			System.out.println(arr[front]+" removed successfully!!!");
			front++;
		}
	}
	
	void print() {
		if(front == -1 && rear == -1)
			System.out.println("Queue is empty!!!");
		else if(front>rear) {
			System.out.println("Queue is empty!!!");
		}else {
			for(int i=front;i<=rear;i++) {
				System.out.print(arr[i]+" ");
			}
			System.out.println();
		}
		
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the size of queue:");
		int size = sc.nextInt();
		Queue q = new Queue(size);
		int data,choice=0;
		
		while(choice != 4) {
			System.out.println("1. Enqueue\n2. Dequeue\n3. Print\n4. Exit");
			System.out.println("Enter your choice:");
			choice = sc.nextInt();
			switch(choice) {
			case 1:
				System.out.println("Enter the data:");
				data = sc.nextInt();
				q.enqueue(data);
				break;
			case 2:
				q.dequeue();
				break;
			case 3:
				q.print();
				break;
			case 4:
				System.exit(0);
				break;
			default:
				System.out.println("Enter valid choice");
			}
		}
		sc.close();
	}
}